DuckDBPyRelation: support `offset` argument for `limit()`

wisp3rwind opened the pull request:
After @hannes suggestion in https://github.com/duckdb/duckdb/discussions/3711, I tried to add the `offset` argument, which indeed seems to be pretty straightforward. I didn't test this in any way, but building (`BUILD_PYTHON=1`) locally succeeds (except for the fact that `make debug` also tries to install the python package, which fails due to permission errors).

Does this look right? Or did I miss other places in the code that need to be adapted?

wisp3rwind committed:
DuckDBPyRelation: support `offset` argument for `limit()`

pdet commented on tools/pythonpkg/src/include/duckdb_python/pyrelation.hpp:
```diff
@@ -93,7 +93,7 @@ struct DuckDBPyRelation {
 	                                             DuckDBPyConnection *conn = DuckDBPyConnection::DefaultConnection());
 
-	unique_ptr<DuckDBPyRelation> Limit(int64_t n);
+	unique_ptr<DuckDBPyRelation> Limit(int64_t n, int64_t offset = 0);
```
This should be an unsigned, no?

wisp3rwind replied:
The main C++ code
https://github.com/duckdb/duckdb/blob/ac159c1673f87f0cd489a12eb024c446e7ace285/src/include/duckdb/main/relation.hpp#L86
uses an `int64_t`, which is why I chose the same here. Negative values seem to be ignored in https://github.com/duckdb/duckdb/blob/master/src/main/relation/limit_relation.cpp

pdet commented:
Thanks for the PR, looks good, just a small nitpicking comment :-)

wisp3rwind commented:
Oh, and don't merge this yet, I didn't update `duckdb-stubs`.

pdet commented:
Fair enough.
Regarding the stubs I think we gave up on maintaining them, Im not sure if we should remove them completely or go back on supporting them. @Mytherin any thoughts?

Alex-Monahan commented:
> Fair enough. Regarding the stubs I think we gave up on maintaining them, Im not sure if we should remove them completely or go back on supporting them. @Mytherin any thoughts?

Are the stubs what provide autocomplete in an IDE? If so, they are awesome to have!

Mytherin commented:
If Alex thinks it's important I think we should keep the stubs then :)

@wisp3rwind Could you update the stubs as part of this PR please?

wisp3rwind committed:
tools: Fix name of regenerate_python_stub.sh script in a comment

wisp3rwind commented:
> @wisp3rwind Could you update the stubs as part of this PR please?

I didn't manage to make the `regenerate_python_stubs` script work, so I just updated the stub manually.

If stubs are considered important after all, I think it would be great to revive the stub test, which was removed in https://github.com/duckdb/duckdb/pull/3221.

pdet commented:
Hi @wisp3rwind, we had some internal discussions and decided to maintain the stubs. 
Would you like to bring the tests back in this PR? If not I can do it later on.

Mytherin commented:
Could you perhaps merge with master and have another look at the stubs now that #3777 has been merged?

wisp3rwind commented:
> Could you perhaps merge with master and have another look at the stubs now that #3777 has been merged?

Rebased on master & force-pushed now.

> I didn't manage to make the regenerate_python_stubs script work, so I just updated the stub manually.

Still applies; I've not tried to run the script again.

Mytherin commented:
Thanks! LGTM

@pdet can this be merged?

pdet commented:
LGTM

Mytherin commented:
Thanks!
